lib/puppet/parser/functions/systemd_escape.rb

```diff
@@ -0,0 +1,15 @@
+require 'puppet/util/execution'
+
+module Puppet::Parser::Functions
+  newfunction(:systemd_escape, :type => :rvalue, :doc => <<-EOS
+  Returns a file system path as escaped by systemd.
+  https://www.freedesktop.org/software/systemd/man/systemd.unit.html#String%20Escaping%20for%20Inclusion%20in%20Unit%20Names
+  EOS
+  ) do |args|
+    raise Puppet::ParseError, ("validate_cmd(): wrong number of arguments (#{args.length}; must be 1)") if args.length != 1
```

Comment:
I'm guessing this was copied from the `validate_cmd` function code. It should be adapted to reflect the name of the function.

Reply:
nitpick: It isn't the fault from the systemd team:
* https://www.archlinux.org/news/binaries-move-to-usrbin-requiring-update-intervention/
* https://lists.archlinux.org/pipermail/arch-dev-public/2012-March/022625.html